packages/ti_rapid7_threat_command/data_stream/ioc/elasticsearch/ingest_pipeline/default.yml

```
@@ -311,6 +343,65 @@ processors:
           return;
         }
       ignore_failure: true
+  - script:
+      lang: painless
+      tag: script-default-deleted_at
+      if: ctx.rapid7?.tc?.ioc?.deleted_at == null && ctx._conf?.ioc_expiration_duration != null && ctx._conf.ioc_expiration_duration != ''
+      description: Indicator Expiration is done after `_conf.ioc_expiration_duration` (default 90d) since its creation time. This script adds a default `rapid7.tc.ioc.deleted_at` field to allow indicator expiration.
+      source: >
+        ZonedDateTime _tmp_deleted_at;
+        ZonedDateTime _tmp_last_seen;
+        if (ctx.rapid7?.tc?.ioc?.status != null && ctx.rapid7.tc.ioc.status.toLowerCase().contains('retire')) {
+          _tmp_deleted_at = ZonedDateTime.parse(ctx['@timestamp']);
+        }
+        else {
+            def dur = ctx._conf?.ioc_expiration_duration;
```

```suggestion
            def dur = ctx._conf.ioc_expiration_duration;
```
Already known to be non-null.

---

updated as suggested